sources/common-dylan/byte-vector.dylan

```diff
@@ -221,3 +221,42 @@ define sealed inline method byte-storage-offset-address
            (primitive-repeated-slot-as-raw
               (the-buffer, primitive-repeated-slot-offset(the-buffer)))))
 end method;
+
+define method from-hexstring (string :: <byte-string>)
```

Comment:
The reason for using <byte-string> instead of <string> is efficiency - I originally used <string> but @waywardmonkeys instructed me to try changing it to <byte-string>.

The resulting DFM & generated C code is much more efficient, having fewer dispatches.